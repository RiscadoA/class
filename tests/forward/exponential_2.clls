proc main() {
    cut {
        println("!"); !a(x: close);
        println("close"); close x
        |a: ?wait|
        println("? 1"); ?a;
        par {
            println("fwd"); fwd! b a
            ||
            println("call 2"); call a(y);
            println("wait 2"); wait y;
            ()
        }
        |b: ?wait|
        println("? 2"); ?b;
        println("call 1"); call b(z);
        println("wait 1"); wait z;
        ()
    }
};;
