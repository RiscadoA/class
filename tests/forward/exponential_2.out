!
close
? 1
fwd
? 2
call 1
wait 1
call 2
wait 2
